.github/workflows/integ_test_analytics.yml

```diff
@@ -24,88 +24,23 @@ permissions:
-        with:
-          project_path: ./AmplifyPlugins/Analytics/Tests/AnalyticsHostApp
-          scheme: AWSPinpointAnalyticsPluginIntegrationTests
-          destination: 'platform=iOS Simulator,name=iPhone 14,OS=latest'
-          xcode_path: '/Applications/Xcode_14.3.app'
-
-  analytics-integration-test-tvOS:
-    if: ${{ github.event_name != 'workflow_dispatch' || inputs.tvos }}
-    runs-on: macos-13
-    environment: IntegrationTest
-    steps:
-      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
-        with:
-          persist-credentials: false
-
-      - name: Make directory
-        run: mkdir -p ~/.aws-amplify/amplify-ios/testconfiguration/
-
-      - name: Copy integration test resouces
-        uses: ./.github/composite_actions/download_test_configuration
-        with:
-          resource_subfolder: analytics
-          aws_role_to_assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
-          aws_region: ${{ secrets.AWS_REGION }}
-          aws_s3_bucket: ${{ secrets.AWS_S3_BUCKET_INTEG_V2 }}
-
-      - name: Run Integration test
-        uses: ./.github/composite_actions/run_xcodebuild_test
-        with:
-          project_path: ./AmplifyPlugins/Analytics/Tests/AnalyticsHostApp
-          scheme: AWSPinpointAnalyticsPluginIntegrationTests
-          destination: platform=tvOS Simulator,name=Apple TV 4K (3rd generation),OS=latest
-          sdk: appletvsimulator
-          xcode_path: '/Applications/Xcode_14.3.app'
-
-  analytics-integration-test-watchOS:
-    if: ${{ github.event_name != 'workflow_dispatch' || inputs.watchos }}
-    runs-on: macos-13
-    environment: IntegrationTest
-    steps:
-      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
-        with:
-          persist-credentials: false
-
-      - name: Make directory
-        run: mkdir -p ~/.aws-amplify/amplify-ios/testconfiguration/
-
-      - name: Copy integration test resouces
-        uses: ./.github/composite_actions/download_test_configuration
-        with:
-          resource_subfolder: analytics
-          aws_role_to_assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
-          aws_region: ${{ secrets.AWS_REGION }}
-          aws_s3_bucket: ${{ secrets.AWS_S3_BUCKET_INTEG_V2 }}
-
-      - name: Run Integration test
-        uses: ./.github/composite_actions/run_xcodebuild_test
-        with:
-          project_path: ./AmplifyPlugins/Analytics/Tests/AnalyticsHostApp
-          scheme: AWSPinpointAnalyticsPluginIntegrationTestsWatch
-          destination: platform=watchOS Simulator,name=Apple Watch Series 8 (45mm),OS=latest
-          sdk: watchsimulator
-          xcode_path: '/Applications/Xcode_14.3.app'
+  analytics-integration-tests:
+    name: Analytics Integration Tests
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [iOS, tvOS, watchOS]
+        exclude:
+          - platform: ${{ github.event.inputs.ios == 'false' && 'iOS' || 'Dummy' }}
```

Comment:
what is the `Dummy` value?

Reply:
It's meant to be a non-existing value that is used if `github.event.inputs.ios == 'false'` is `false`, i.e. if we haven't explicitly disabled running for iOS.

Basically it's saying "exclude the **Dummy** platform from the matrix", but since **Dummy** is not in there, nothing happens.

I could probably have used another name, like **None** or something 😅

Reply:
Got it, we can do a follow up PR with the `None` changes.